java/ql/test/library-tests/frameworks/JaxWs/JakartaRs1.java

```diff
@@ -0,0 +1,196 @@
+import jakarta.ws.rs.PUT;
+import jakarta.ws.rs.OPTIONS;
+import jakarta.ws.rs.HEAD;
+import jakarta.ws.rs.Path;
+import jakarta.ws.rs.BeanParam;
+import jakarta.ws.rs.CookieParam;
+import jakarta.ws.rs.FormParam;
+import jakarta.ws.rs.HeaderParam;
+import jakarta.ws.rs.MatrixParam;
+import jakarta.ws.rs.PathParam;
+import jakarta.ws.rs.Produces;
+import jakarta.ws.rs.QueryParam;
+import jakarta.ws.rs.client.Client;
+import jakarta.ws.rs.core.Context;
+import jakarta.ws.rs.core.MediaType;
+import jakarta.ws.rs.core.MultivaluedMap;
+import jakarta.ws.rs.core.Response;
+import jakarta.ws.rs.ext.MessageBodyReader;
+
+@Path("")
+public class JakartaRs1 { // $RootResourceClass
+  public JakartaRs1() { // $InjectableConstructor
+  }
+
+  @GET
+  int Get() { // $ResourceMethod $ResourceMethodOnResourceClass
+    return 0; // $XssSink
+  }
+
+  @POST
+  void Post() { // $ResourceMethod $ResourceMethodOnResourceClass
+  }
+
+  @Produces("text/plain") // $ProducesAnnotation=text/plain
+  @DELETE
+  double Delete() { // $ResourceMethod=text/plain $ResourceMethodOnResourceClass
+    return 0.0; // $XssSink
+  }
+
+  @Produces(MediaType.TEXT_HTML) // $ProducesAnnotation=text/html
+  @PUT
+  void Put() { // $ResourceMethod=text/html $ResourceMethodOnResourceClass
+  }
+
+  @OPTIONS
+  void Options() { // $ResourceMethod $ResourceMethodOnResourceClass
+  }
+
+  @HEAD
+  void Head() { // $ResourceMethod $ResourceMethodOnResourceClass
+  }
+
+  @Path("")
+  NonRootResourceClassJakarta subResourceLocator() { // $SubResourceLocator
+    return null;
+  }
+
+  public class NonRootResourceClassJakarta { // $NonRootResourceClass
+    @GET
+    int Get() { // $ResourceMethod $ResourceMethodOnResourceClass
+      return 0; // $XssSink
+    }
+
+    @Produces("text/html") // $ProducesAnnotation=text/html
+    @POST
+    boolean Post() { // $ResourceMethod=text/html $ResourceMethodOnResourceClass
+      return false;
+    }
+
+    @Produces(MediaType.TEXT_PLAIN) // $ProducesAnnotation=text/plain
+    @DELETE
+    double Delete() { // $ResourceMethod=text/plain $ResourceMethodOnResourceClass
+      return 0.0; // $XssSink
+    }
+
+    @Path("")
+    AnotherNonRootResourceClassJakarta subResourceLocator1() { // $SubResourceLocator
+      return null;
+    }
+
+    @GET
+    @Path("")
+    NotAResourceClass1Jakarta NotASubResourceLocator1() { // $ResourceMethod $ResourceMethodOnResourceClass
+      return null; // $XssSink
+    }
+
+    @GET
+    NotAResourceClass2Jakarta NotASubResourceLocator2() { // $ResourceMethod $ResourceMethodOnResourceClass
+      return null; // $XssSink
+    }
+
+    NotAResourceClass2Jakarta NotASubResourceLocator3() {
+      return null;
+    }
+  }
+}
+
+class AnotherNonRootResourceClassJakarta { // $NonRootResourceClass
+  public AnotherNonRootResourceClassJakarta() {
+  }
+
+  public AnotherNonRootResourceClassJakarta(@BeanParam int beanParam, @CookieParam("") int cookieParam, @FormParam("") int formParam, // $InjectionAnnotation
```

Comment:
Given that we're testing parameter annotations here with line-granularity, I think it would be good to put each parameter on its own line (e.g. newline after each `,`), so it's clear what the test expectations refer to.